### PR TITLE
[ENH] Migrate exercise and solutions: Part 1

### PR DESCRIPTION
**Pull request by HumphreyYang**
Hi @jstac and @mmcky,

I have migrated the first part of the lecture to use sphinx-exercise for the first section: Tools and Techniques.

Here is a summary of places to look at when reviewing this PR: 

1. I changed the _config.yml to enable spinx-exercise and drop-down solutions;
2. In lecture [4. Discrete State Dynamic Programming](https://63077285fc58d3007638c3a0--wonderful-lalande-528d1c.netlify.app/discrete_dp.html#id11), I think the exercise in it is more like a worked example rather than for students to practice themself. Thus, I did not change the style of this exercise. Please kindly let me know if you think it should also be migrated to sphinx-exercise (I think sphinx-exercise does not support adding sections within it as well);
3. A warning and a potential typo were fixed. Please see the following comments for details.

Could you please kindly review whether these changes meet your expectation? If it looks good to you, I will use these as a template for further migration. 

Many thanks.

### CODE REVIEW

File: lectures/muth_kalman.md [hunk 1]
```diff
@@ -237,10 +237,11 @@ simulate it
...
                [K1, 1 - K1, K1 * σ_y],
-               [ 0,      0,        0]])
+               [ 0,      0,        0]],dtype=object)
 Cf = np.array([[σ_x,        0],
                [  0, K1 * σ_y],
-               [  0,        1]])
+               [  0,        1]],
+               dtype=object)
 Gf = np.array([[1,  0, σ_y],
                [1, -1, σ_y]])
```
Comment (HumphreyYang) [inline lectures/muth_kalman.md:247]: addressed a warning by adding dtype = object;

File: lectures/stationary_densities.md [hunk 2]
```diff
@@ -984,7 +1010,7 @@ series for one boxplot all at once
 ```{code-cell} python3
 n = 20
 k = 5000
-J = 6
+J = 8
```
Comment (HumphreyYang) [inline lectures/stationary_densities.md:1013]: I think J should = 8 given the set-up of the exercise

**Comment by jstac** [comment]
Thanks @HumphreyYang .

The only thing I'm not sure about is the `dtype=object` change.  That array looks like it only contains numerical data, in which case we shouldn't be setting it to `object` (which will prevent fast execution). Is there some other way around the warning?

Looking at the code, I can't see why that warning is occuring.  

Also, I recommend that you create an issue with a checkbox list for converting the exercises and solutions in each lecture.  https://docs.github.com/en/issues/tracking-your-work-with-issues/about-task-lists    Each time you finish the conversions for a lecture, you can tick it off.

**Comment by HumphreyYang** [comment]
Hi @jstac,
Thanks for your review.

> Looking at the code, I can't see why that warning is occurring.
I reckon it is because K1 is a nparray from the kmuth.stationary_values() function.
```
In [13]: K1
Out[13]: array([[0.18099751]])
```
I can get around by double indexing K1[0][0] but it makes the code less readable. I can also do this double indexing when K1 is first assigned.

> Also, I recommend that you create an issue with a checkbox list for converting the exercises and solutions in each lecture. https://docs.github.com/en/issues/tracking-your-work-with-issues/about-task-lists Each time you finish the conversions for a lecture, you can tick it off.

I will create the tick box now. Thanks for your suggestion.

**Comment by jstac** [comment]
Thanks @HumphreyYang 

I think your double indexing suggestion is the right one.  (Either that or the Kalman filter code should be adusted to return a scalar value in the 1 x 1 case.)

**Comment by HumphreyYang** [comment]
Hi @jstac,

I have fixed the warning using double indexing. Please kindly check the latest deployment and merge if it looks good to you now.

Many thanks.

**Comment by jstac** [comment]
Thanks @HumphreyYang .

Since many look ups of K1[0][0] is not efficient, it might be best to write k1 = K1[0][0] above and then use k1.

**Comment by HumphreyYang** [comment]
Hi @jstac,

I have updated S1 and K1. Please kindly check the latest [deployment](https://630dbc35cc656e13530812ba--wonderful-lalande-528d1c.netlify.app/muth_kalman.html#id8) and merge if it looks good to you.

**Comment by jstac** [comment]
Nice work @HumphreyYang , many thanks.  Merging.

**Pull request merged**